mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/DeletedMessage.java

```diff
@@ -141,7 +141,7 @@ public FinalStage(MessageId messageId, List<MailboxId> originMailboxes, User own
             }
 
             public FinalStage subject(String subject) {
-                this.subject = Optional.of(subject);
+                this.subject = Optional.ofNullable(subject);
```

Comment:
Why is this change needed?

Reply:
there is no guarantee that the caller will pass a non null subject to `subject(String subject)` method. In that case it throws an unexpected error.

Reply:
I would argue that it throws an expected `NullPointerException` that is just not yet tested and all of this "fails early" classical tactic prevents later unexpected behaviour.

Let's be "null free" whereever we can. New APIs definitely are such places.

Maybe what you truely want is to pass the `subject` as an optional like I did here: https://github.com/linagora/james-project/pull/2194/commits/ad0550fe869ff80a4bbbd5fd3a15c47594a83fe7

Reply:
okay, I revered to use the original builder method.